MailContainerTest.Tests/MailTransferServiceTests.cs

```diff
@@ -0,0 +1,61 @@
+using MailContainerTest.Data;
+using MailContainerTest.Services;
+using MailContainerTest.Types;
+using Moq;
+using NUnit.Framework;
+
+namespace MailContainerTest.Tests
+{
+    public class MailTransferServiceTests
+    {
+        [Test]
+        public void MakeMailTransfer_TransferAllowed_ContainersAreUpdated()
+        {
+            var dataStore = new Mock<IMailContainerDataStore>();
+
+            dataStore
+                .Setup(m => m.GetMailContainer(It.IsAny<string>()))
+                .Returns(new MailContainer());
+
+            var requestApprovalService = new Mock<IMakeMailTransferRequestApprovalService>();
+
+            requestApprovalService
+                .Setup(m => m.TransferIsAllowed(It.IsAny<MailContainer>(), It.IsAny<MakeMailTransferRequest>()))
+                .Returns(true);
+
+            var transferService = new MailTransferService(dataStore.Object, requestApprovalService.Object);
+
+            var result = transferService.MakeMailTransfer(new MakeMailTransferRequest());
+
+            dataStore.Verify(
```

Comment:
Here I would verify whether the source and destination containers were each updated once, rather than checking a total of two updates across any containers.